accepted/00000-saltclass-external-pillar.md

```diff
@@ -0,0 +1,147 @@
+- Feature Name: Saltclass as external pillar mechanism
+- Start Date: 2019-08-20
+- RFC PR:
+
+# Summary
+[summary]: #summary
+
+Enhance/replace current custom external pillar mechanism with upstream provided Saltclass module.
+
+# Motivation
+[motivation]: #motivation
+
+SUSE Manager needs a way how to export data about systems, groups, services etc. to the salt configuration management. In current implementation this is done by custom python script [suma_minion.py](https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/modules/pillar/suma_minion.py) used by salt master leveraging its external pillar mechanism. This worked so far very well, even allowing Uyuni some extra tricks like integration of Formulas with Forms mechanism. However the implementation has also some limitations preventing further growth.
+
+This RFC proposes to, at first, enable yet another external pillar mechanism in the salt, but this time not a custom script, but salt's own [saltclass](https://docs.saltstack.com/en/latest/ref/pillar/all/salt.pillar.saltclass.html) mechanism.
+
+Examples of limitations of our external pillar implementation:
+- allows only static pillars (no jinja preparsing, no pillar dependencies on other pillars)
+- problematic data merging
+- made integration of formulas easy, but causes duplication of code where i.e. form.yml is parsed on both java and python level
+
+# Detailed design
+[design]: #detailed-design
+
+There are three choices how to address the introduction of saltclass to the SUSE Manager:
+
+A) Enable concurrent usage of both custom `suma_minion.py` and `saltclass` external pillars
+B) Enable concurrent usage of both custom `suma_minion.py` and `saltclass` external pillars, but with the target to gradually migrate from `suma_minion.py` to `saltclass`.
+C) Do the one time migration and switch from custom `suma_minion.py` to `saltclass`
+
```

Comment:
I'd strongly prefer C).

The current code is convoluted enough. It usually takes me a few minutes to find out how, for example, the pillar data indirectly drives formula inclusion via the master_tops file etc.

Adding yet another concept without removing this mess would make it even harder for customers to extend Manager in a meaningful way.